### PR TITLE
[WIP] Batch Trade Fixes

### PR DESCRIPTION
**Pull request by controtie**

### CODE REVIEW

File: src/api/utils/tradeQuoter.ts [hunk 1]
```diff
@@ -95,12 +95,18 @@ export class TradeQuoter {
...
+    // This value is already floored. Number that ends in 8, should end in .7
     const fromTokenRequestAmount = this.calculateFromTokenAmount(
       setOnChainDetails,
       fromTokenAddress,
       amount
     );
 
+    const positionForFromToken = setOnChainDetails
+      .positions
+      .find((p: any) => p.component.toLowerCase() === fromTokenAddress.toLowerCase());
```
Comment (controtie) [inline src/api/utils/tradeQuoter.ts:107]: This is the true final amount that we want to submit for quote validation & return to the front-end, right?

File: src/api/utils/tradeQuoter.ts [hunk 2]
```diff
@@ -312,8 +320,14 @@ export class TradeQuoter {
       (feePercentage / 100)
     );
 
+
     const fromTokenAmount = quote.sellAmount;
 
+    let fromUnits;
+    if (fromTokenRequestAmount.eq(fromTokenAmount)) {
```
Comment (asoong) [inline src/api/utils/tradeQuoter.ts:327]: You cannot use `fromTokenRequestAmount`. Compare `fromTokenAmount` to the max value implied by the position. If so, return the position's `units` which are then passed to the TradeModule (you are trying to trade the entire position but can't account for the precision)

File: src/api/utils/tradeQuoter.ts [hunk 3]
```diff
@@ -392,6 +407,7 @@ export class TradeQuoter {
       throw new Error('Invalid fromToken input');
     }
 
+    // This logic needs to be moved/present in `fetchZeroExQuoteForTradeModule`
     const totalSupply = setOnChainDetails.totalSupply;
     const impliedMaxNotional = positionForFromToken.unit.mul(totalSupply).div(SCALE);
```
Comment (controtie) [inline src/api/utils/tradeQuoter.ts:412]: Need to move this into `fetchZeroExTradeQuote` method

**Comment by controtie** [comment]
Solved in #118

**Pull request closed without merging**